src/aro/Parser.zig

```diff
@@ -6823,7 +6825,21 @@ pub const Result = struct {
             }
         } else {
             if (c == .assign and (dest_unqual.is(p.comp, .array) or dest_unqual.is(p.comp, .func))) {
-                try p.errTok(.not_assignable, tok);
+                const type_name = blk: {
+                    const strings_top = p.strings.items.len;
+                    defer p.strings.items.len = strings_top;
+
+                    const writer = p.strings.writer();
+
+                    try dest_unqual.print(p.comp, writer);
+
+                    break :blk try p.comp.diagnostics.arena.allocator().dupe(u8, p.strings.items[strings_top..]);
+                };
+                try p.errExtra(switch (dest_unqual.type(p.comp)) {
```

Comment:
This should use `base()` since a `is(.array` returns true for a `typedef(array)` . I need to add some documentation to these, sorry about that.